flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/windowing/functions/InternalWindowFunction.java

```diff
@@ -46,5 +46,5 @@
 	 *
 	 * @throws Exception The function may throw exceptions to fail the program and trigger recovery.
 	 */
-	public abstract void apply(KEY key, W window, IN input, Collector<OUT> out) throws Exception;
+	public abstract void process(KEY key, W window, IN input, Collector<OUT> out) throws Exception;
```

Comment:
I think this PR is good as a first step but I really want to see the benefits of introducing the new interface, especially for users. I don't see how the window metadata is extended with the current implementation.  That's my $0.02. @aljoscha is more of an expert here.